### PR TITLE
fix: support all allowed protocol numbers

### LINKED ISSUE #111517: [Community ID Processor] Handle events without a named network.transport
### Elasticsearch Version

8.14.1

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

docker.elastic.co/elasticsearch/elasticsearch:8.14.1

### Problem Description

The community_id processor [assumes](https://github.com/elastic/elasticsearch/blob/c5da25754007af51029a676a8784b16be16d8995/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/CommunityIdProcessor.java#L226) that it can always determine the transport name from the [IANA number](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml). But this is not true and should not be required to compute a [community_id](https://github.com/corelight/community-id-spec?tab=readme-ov-file#technical-details).

It should be sufficient to know that the transport is not TCP / UDP / SCTP / ICMPv4 / ICMPv6, and then proceed to a computation using only the source ip, destination ip, and protocol number.

## Related

- https://github.com/elastic/integrations/issues/10678

### Steps to Reproduce

```
POST _ingest/pipeline/_simulate
{
  "pipeline": {
    "processors": [
    {
      "community_id" : {}
    }
    ]
  },
  "docs": [
    {
      "_source": {
        "source": {"ip":"192.168.1.2"},
        "destination": {"ip": "10.1.2.3"},
        "network": {"iana_number": "4"}
      }
    }
  ]
}
```

What I got:

```
{
  "docs": [
    {
      "error": {
        "root_cause": [
          {
            "type": "illegal_argument_exception",
            "reason": "unknown transport protocol number [4]"
          }
        ],
        "type": "illegal_argument_exception",
        "reason": "unknown transport protocol number [4]"
      }
    }
  ]
}
```

What I expected:

```json
{
  "destination": {
    "ip": "10.1.2.3"
  },
  "network": {
    "community_id": "1:KXQzmk3bdsvD6UXj7dvQ4bM6Zvw=",
    "iana_number": "4"
  },
  "source": {
    "ip": "192.168.1.2"
  }
}
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
This PR implements support in community_id processor to produce a respective hash for protocol numbers that are valid but not in the predefined list that require special handling 

Closes https://github.com/elastic/elasticsearch/issues/111517

### CODE REVIEW

**Comment by masseyke** [comment]
Thanks for the PR! Could you also update the user-facing documentation about `iana_number`? I'm not sure how best to phrase it, but it is no longer fully true after this PR that those are the only ones we support.

**Comment by pkoutsovasilis** [comment]
> Thanks for the PR! Could you also update the user-facing documentation about `iana_number`? I'm not sure how best to phrase it, but it is no longer fully true after this PR that those are the only ones we support.

thanks for catching this @masseyke 🙂 I did rephrase it to be truthful

**Comment by pkoutsovasilis** [comment]
@masseyke anything more I need to handle before having a look at this? 🙂

**Comment by masseyke** [comment]
> @masseyke anything more I need to handle before having a look at this? 🙂

No sorry, I'll take a look at this today!

**Review by masseyke — APPROVED** [review]
LGTM

**Comment by aleksmaus** [comment]
strangely I'm seeing now different results with for example ```cisco_ios``` integration on main branch when regenerating the pipeline results against 8.15 and 8.16 snapshots of the stack. 
@pkoutsovasilis is this expected?
